### PR TITLE
Maurits/fair checklist

### PR DESCRIPTION
**Pull request by mwakok**
Checklist to improve the FAIR principles for research software. Preferably, I would like to have this available before we send out all review summaries so we can include the link.

This is a first attempt, any suggestions are welcome.

### CODE REVIEW

File: docs/03_Software/07_FAIR_checklist/checklist.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+#### Code quality
+_Essential_
+- [ ] Project [organisation](https://coderefinery.github.io/reproducible-research/02-organizing-projects/)
+- [ ] Record [software dependencies](https://coderefinery.github.io/reproducible-research/03-dependencies/)
+
+_Recommended_
+- [ ] Follow [PEP8 guidelines](https://realpython.com/python-pep8/)
+- [ ] Use linter (e.g. [pylint](https://pypi.org/project/pylint/), [flake8](https://pypi.org/project/flake8/))
+- [ ] Use a formatter (e.g. [black](https://github.com/psf/black), [yapf](https://github.com/google/yapf))
+- [ ] Follow [best coding practices](https://alan-turing-institute.github.io/rse-course/html/index.html)
```
Comment (niketagrawal) [inline docs/03_Software/07_FAIR_checklist/checklist.md:78]: This is the more precise link with a list of recommendations https://alan-turing-institute.github.io/rse-course/html/module07_construction_and_design/07_03_refactoring.html

File: docs/03_Software/07_FAIR_checklist/checklist.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+- [ ] Build [API reference](https://developer.lsst.io/python/numpydoc.html) from docstrings
+
+
+#### Testing
+_Recommended_  
+- [ ] [Defensive programming](https://swcarpentry.github.io/python-novice-inflammation/10-defensive/index.html)
+- [ ] Integration or build test
+- [ ] User installation test
+- [ ] Unit tests (coverage >70%)
+- [ ] Continuous integration (GitHub Actions)
```
Comment (niketagrawal) [inline docs/03_Software/07_FAIR_checklist/checklist.md:53]: Add the below URL for information on Continuous Integration from the Turing way guide.
https://the-turing-way.netlify.app/reproducible-research/ci/ci-options.html

File: docs/03_Software/07_FAIR_checklist/checklist.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+- [ ] Website ([github.io pages](https://pages.github.com/), [Readthedocs](https://readthedocs.org/), [JupyterBook](https://jupyterbook.org/intro.html)
+- [ ] Build [API reference](https://developer.lsst.io/python/numpydoc.html) from docstrings
+
+
+#### Testing
+_Recommended_  
+- [ ] [Defensive programming](https://swcarpentry.github.io/python-novice-inflammation/10-defensive/index.html)
+- [ ] Integration or build test
+- [ ] User installation test
+- [ ] Unit tests (coverage >70%)
```
Comment (niketagrawal) [inline docs/03_Software/07_FAIR_checklist/checklist.md:52]: Add the below URL for information on unit testing from the Turing way guide.
https://the-turing-way.netlify.app/reproducible-research/testing/testing-unittest.html

File: docs/03_Software/07_FAIR_checklist/checklist.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+_Recommended_  
+- [ ] Examples and tutorials (Jupyter notebooks, videos, screencasts)
+- [ ] Website ([github.io pages](https://pages.github.com/), [Readthedocs](https://readthedocs.org/), [JupyterBook](https://jupyterbook.org/intro.html)
+- [ ] Build [API reference](https://developer.lsst.io/python/numpydoc.html) from docstrings
+
+
+#### Testing
+_Recommended_  
+- [ ] [Defensive programming](https://swcarpentry.github.io/python-novice-inflammation/10-defensive/index.html)
+- [ ] Integration or build test
```
Comment (niketagrawal) [inline docs/03_Software/07_FAIR_checklist/checklist.md:50]: Add the below URL for information on integration testing from the Turing way guide.
https://the-turing-way.netlify.app/reproducible-research/testing/testing-integrationtest.html

File: docs/03_Software/07_FAIR_checklist/checklist.md [hunk 5]
```diff
@@ -0,0 +1,94 @@
+# FAIR for research software guidelines
+
+To ensure your research software is findable and usable by other researchers, we have compiled a checklist to improve its FAIRness (Findable, Accessible, Interoperable, Reproducible).
```
Comment (niketagrawal) [inline docs/03_Software/07_FAIR_checklist/checklist.md:3]: *Reusable

File: docs/03_Software/07_FAIR_checklist/checklist.md (hunk 1 above)
Comment (mwakok) [reply docs/03_Software/07_FAIR_checklist/checklist.md:78]: Good point, but then I would use a link to the entire chapter, as that covers other best-practices as well: https://alan-turing-institute.github.io/rse-course/html/module07_construction_and_design/index.html

File: docs/03_Software/07_FAIR_checklist/checklist.md (hunk 3 above)
Comment (mwakok) [reply docs/03_Software/07_FAIR_checklist/checklist.md:52]: I have combined the check on unit and integration tests, and added your suggested links

**Review by niketagrawal — APPROVED** [review]
Thanks for putting together the checklist and incorporating the review comments.

**Comment by jurra** [comment]
One thing would add though is a brief paragraph introducing when and what for to use the checklist.
That is not yet there at the moment ;)

**Pull request merged**